### PR TITLE
Run linter and phpunit test with php7.3

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Prepare ownCloud for php 7.3

## Motivation and Context
Ride on the tip of the wave ...

## How Has This Been Tested?
- :robot: 

## ToDo
- [ ] sabre/libs are not ready ... working on this ....

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x]] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by PVince81** [comment]
styler not happy:
```
   1) console.php (native_function_invocation)
      ---------- begin diff ----------
--- Original
+++ New
@@ -50,7 +50,7 @@
 
 // running oC on Windows is unsupported since 8.1, this has to happen here because
 // is seems that the autoloader on Windows fails later and just throws an exception.
-if (stripos(PHP_OS, 'WIN') === 0) {
+if (\stripos(PHP_OS, 'WIN') === 0) {
 	echo 'ownCloud Server does not support Microsoft Windows.';
 	exit(1);
 }
@@ -97,7 +97,7 @@
 		exit(1);
 	}
 
-	if (!\function_exists('pcntl_signal') && !in_array('--no-warnings', $argv, true)) {
+	if (!\function_exists('pcntl_signal') && !\in_array('--no-warnings', $argv, true)) {
 		echo 'The process control (PCNTL) extensions are required in case you want to interrupt long running commands - see http://php.net/manual/en/book.pcntl.php' . PHP_EOL;
 	}
 

      ----------- end diff -----------

   2) index.php (native_function_invocation)
      ---------- begin diff ----------
--- Original
+++ New
@@ -44,7 +44,7 @@
 
 // running oC on Windows is unsupported since 8.1, this has to happen here because
 // is seems that the autoloader on Windows fails later and just throws an exception.
-if (stripos(PHP_OS, 'WIN') === 0) {
+if (\stripos(PHP_OS, 'WIN') === 0) {
 	echo 'ownCloud Server does not support Microsoft Windows.';
 	return;
 }

      ----------- end diff -----------


Checked all files in 46.993 seconds, 30.000 MB memory used
Makefile:197: recipe for target 'test-php-style' failed
make: *** [test-php-style] Error 8
```

**Comment by PVince81** [comment]
wow, and everything else is green ?!

**Comment by DeepDiver1975** [comment]
> wow, and everything else is green ?!

no - see https://drone.owncloud.com/owncloud/core/10679/180

sabre libs are not yet ready for php 7.3 - working on this .....

**Comment by DeepDiver1975** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

File: console.php [hunk 1]
```diff
@@ -41,22 +41,22 @@
 	exit(1);
 }
 
-// Show warning if PHP 7.3 is used as ownCloud is not compatible with PHP 7.3
-if (\version_compare(PHP_VERSION, '7.3.0alpha1') !== -1) {
-	echo 'This version of ownCloud is not compatible with PHP 7.3' . PHP_EOL;
+// Show warning if PHP 8 is used as ownCloud is not compatible with PHP 8
+if (\version_compare(PHP_VERSION, '8.0.0alpha1') !== -1) {
```
Comment (patrickjahns) [inline console.php]: 7.4 ?

Comment (DeepDiver1975) [reply console.php]: There will be No 7.4
7.3 is the Last of the 7 releases

Comment (DeepDiver1975) [reply console.php]: I was confused - yes 7.4 - fixed - thanks for spotting

**Comment by DeepDiver1975** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 2]
```diff
@@ -1009,7 +1009,7 @@ public function theResponseShouldContainACustomPropertyWithValue(
 	public function asTheFileOrFolderShouldNotExist($user, $entry, $path) {
 		$client = $this->getSabreClient($user);
 		$response = $client->request(
-			'HEAD', $this->makeSabrePath($user, $path)
+			'GET', $this->makeSabrePath($user, $path)
```
Comment (PVince81) [inline tests/acceptance/features/bootstrap/WebDav.php]: behavior change ?

File: apps/dav/lib/Avatars/AvatarHome.php [hunk 3]
```diff
@@ -96,7 +95,7 @@ public function delete() {
 	}
 
 	public function getName() {
-		list(, $name) = URLUtil::splitPath($this->principalInfo['uri']);
+		list(, $name) = \Sabre\Uri\split($this->principalInfo['uri']);
```
Comment (PVince81) [inline apps/dav/lib/Avatars/AvatarHome.php]: is this an API change ? I suspect other apps will need adjuting as well like custom groups

**Review by PVince81 — APPROVED** [review]
please answer the comments for confirmation then this is good to go if CI passes :+1:

**Comment by phil-davis** [comment]
Will need backport sorted out for supporting PHP 7.3 in `stable10` one day.

**Comment by phil-davis** [comment]
Code for `stable10` is in PRs:
#34559 
#35775

**Pull request merged**